### PR TITLE
For #7059 - Allow search engines to scroll

### PR DESCRIPTION
**Pull request by darkwing**
To test:

0.  Open search by clicking the URL bar from home
1.  Turn your phone horizontally
2.  Scroll the search engines vertically

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by AaronMT** [comment]
@darkwing Do you have access to the test-run videos?

It looks like, on cold start, access to the search engine fragment is entirely empty. Ran against a Nexus 6 25 and a Pixel 2 28.

**Comment by darkwing** [comment]
@AaronMT I can see it failing locally -- I'm checking it out!  Thank you!

File: app/src/main/res/layout/component_awesomebar.xml [hunk 1]
```diff
@@ -6,14 +6,15 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:mozac="http://schemas.android.com/apk/res-auto"
     android:id="@+id/awesomeBar"
+    android:background="@color/photonRed50"
```
Comment (ekager) [inline app/src/main/res/layout/component_awesomebar.xml]: cc @sblatz is this added background color intentional here? Or was it part of testing?

File: app/src/main/res/layout/fragment_search.xml [hunk 2]
```diff
@@ -53,88 +32,126 @@
...
-
-    <androidx.constraintlayout.widget.Barrier
-        android:id="@+id/awesomeBar_barrier"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        app:barrierDirection="bottom"
-        app:constraint_referenced_ids="fill_link_from_clipboard,search_suggestions_onboarding" />
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:id="@+id/scrollable_area"
+            android:background="@color/photonGreen50"
```
Comment (ekager) [inline app/src/main/res/layout/fragment_search.xml]: Same with these background colors added here? I'm a bit confused

**Comment by ekager** [comment]
CC @darkwing @sblatz is this PR ready for review? I was a bit confused by the added background colors

**Comment by darkwing** [comment]
We started this a month ago to ensure the list of search engines could scroll; we ran into one last issue which we haven't figured out yet, and thus the weird colors.  I can pick this up again tomorrow.

**Comment by darkwing** [comment]
The current problem is that there is this odd spacing that shows up when the "Fill link from clipboard" item is present -- it looks great without it:

Screenshot_20200323-101022

Screenshot_20200323-105027

**Comment by ekager** [comment]
Copying from Slack for visibility:
What is `awesomeBar_barrier`? I don't think that `fill_link_from_clipboard` bottom should be constrained to the bottom of it
Removing that line seems to fix the odd space!

**Comment by liuche** [comment]
Hey @darkwing if this is ready, can you re-ping Emily?

**Comment by darkwing** [comment]
Yep, sorry, @ekager this is worth another look!

**Comment by ekager** [comment]
@darkwing could you rebase and fix the conflict ? Can review Monday :)

**Pull request closed without merging**